src/NServiceBus.RavenDB/Internal/DocumentStoreManager.cs

```diff
@@ -0,0 +1,179 @@
+                }
+
+                initContext = holder.DocumentStoreProxy;
+            }
+
+            if (initContext == null)
+            {
+                throw new Exception($"RavenDB is configured as persistence for {typeof(TStorageType).Name} and no DocumentStore instance could be found.");
+            }
+
+            return initContext.Init(settings);
+        }
+
+        private static DocumentStoreInitContext CreateDefaultDocumentStore(ReadOnlySettings settings)
+        {
+            var p = settings.GetOrDefault<ConnectionParameters>(RavenDbSettingsExtensions.DefaultConnectionParameters);
+            if (p != null)
+            {
+                var storeByParams = new DocumentStore
+                {
+                    Url = p.Url,
+                    DefaultDatabase = p.DatabaseName ?? settings.EndpointName(),
+                    ApiKey = p.ApiKey,
+                    Credentials = p.Credentials
+                };
+
+                return new DocumentStoreInitContext(storeByParams);
+            }
+
+            var initContext = CreateStoreByConnectionStringName(settings, "NServiceBus/Persistence/RavenDB", "NServiceBus/Persistence");
+
+            if (initContext != null)
+            {
+                return initContext;
+            }
+
+            return CreateStoreByUrl(settings, "http://localhost:8080");
+        }
+
+        class DocumentStoreHolder
+        {
+            public DocumentStoreInitContext UninitializedDocumentStore { get; set; }
+        }
+
+        static DocumentStoreInitContext CreateStoreByConnectionStringName(ReadOnlySettings settings, params string[] connectionStringNames)
+        {
+            var connectionStringName = GetFirstNonEmptyConnectionString(connectionStringNames);
+            if (!string.IsNullOrWhiteSpace(connectionStringName))
+            {
+                var docStore = new DocumentStore
+                {
+                    ConnectionStringName = connectionStringName
+                };
+                if (docStore.DefaultDatabase == null)
+                {
+                    docStore.DefaultDatabase = settings.EndpointName();
+                }
+
+                return new DocumentStoreInitContext(docStore);
+            }
+            return null;
+        }
+
+        static DocumentStoreInitContext CreateStoreByUrl(ReadOnlySettings settings, string url)
+        {
+            var docStore = new DocumentStore
+            {
+                Url = url
+            };
+
+            if (docStore.DefaultDatabase == null)
+            {
+                docStore.DefaultDatabase = settings.EndpointName();
+            }
+
+            return new DocumentStoreInitContext(docStore);
+        }
+
+        static string GetFirstNonEmptyConnectionString(params string[] connectionStringNames)
+        {
+            try
+            {
+                return connectionStringNames.FirstOrDefault(cstr => ConfigurationManager.ConnectionStrings[cstr] != null);
```

Comment:
The point is to get the first of a list of possible connection string names that are defined. This doesn't actually return a string, it returns a ConnectionStringSettings object (containing Key, Name, ConnectionString, ProviderName).

If there is an _empty_ connection string, I'm sure the database will be more than happy to throw for us.